admin/category_product_listing.php

```diff
@@ -25,1230 +25,1656 @@
+                        : ''; //Disable subcategories or no change?
+                    $products_status = isset($_POST['set_products_status']) && $_POST['set_products_status'] === 'set_products_status_off' ? '0' : ''; //Disable products or no change?
+                } else {//form is coming from a Disabled category which is to be changed to Enabled
+                    $category_status = '1'; //Enable this category
+                    $subcategories_status = isset($_POST['set_subcategories_status']) && $_POST['set_subcategories_status'] === 'set_subcategories_status_on' ? '1'
+                        : ''; //also Enable subcategories or no change?
+                    $products_status = isset($_POST['set_products_status']) && $_POST['set_products_status'] === 'set_products_status_on' ? '1' : ''; //Disable products or no change?
+                }
 
-            if (!empty($cascaded_prod_id_for_delete) && !empty($cascaded_prod_cat_for_delete)) {
-              $product_id = $cascaded_prod_id_for_delete;
-              $product_categories = $cascaded_prod_cat_for_delete;
-              $do_delete_flag = true;
+                for ($i = 0, $n = count($categories); $i < $n; $i++) {
+                    //set categories_status
+                    if ($categories[$i]['id'] === $categories_id) {//always update THIS category
+                        zen_set_category_status($categories[$i]['id'], $category_status);
+                    } elseif ($subcategories_status !== '') {//optionally update subcategories if a change was selected
+                        zen_set_category_status($categories[$i]['id'], $subcategories_status);
+                    }
+
+                    //set products_status
+                    if ($products_status === '') {
+                        continue;
+                    }
+
+                    //only execute if a change was selected
+                    $category_products = zen_get_linked_products_for_category($categories[$i]['id']);
+                    foreach ($category_products as $category_product) {
+                        zen_set_product_status($category_product, $products_status);
+                    }
+                }
+            }
+            zen_redirect(
+                zen_href_link(FILENAME_CATEGORY_PRODUCT_LISTING, 'cPath=' . $_GET['cPath'] . '&cID=' . $_GET['cID'] . (isset($_GET['page']) ? '&page=' . $_GET['page'] : '') . $search_parameter)
+            );
+            break;
+        case 'remove_type':
+            if (isset($_POST['type_id'])) {
+                zen_remove_restrict_sub_categories($_GET['cID'], (int)$_POST['type_id']);
+                $action = 'edit';
+                zen_redirect(zen_href_link(FILENAME_CATEGORY_PRODUCT_LISTING, 'action=edit_category&cPath=' . $_GET['cPath'] . '&cID=' . zen_db_prepare_input($_GET['cID'])));
```

Comment:
$_GET['cID'] is pre-sanitized to an int, no need for zen_db_prepare_input.

Reply:
ok, for next PR.